src/ORM/DataObjectSchema.php

```diff
@@ -666,67 +666,90 @@ protected function buildCustomDatabaseIndexes($class)
 
     protected function buildSortDatabaseIndexes($class)
     {
-        $indexMode = $this->getSortIndexMode($class);
-        if ($indexMode === DataObjectSchema::SORT_INDEX_MODE_NONE) {
+        $sort = Config::inst()->get($class, 'default_sort', Config::UNINHERITED);
+        if (!is_string($sort) && !is_array($sort)) {
+            return [];
+        }
+        return $this->deriveIndexFromSort(
+            DataObjectSchema::tableName($class) ?? '',
+            array_keys($this->databaseFields($class, false)),
+            $sort,
+            $this->getSortIndexMode($class)
+        );
+    }
+
+    /**
+     * Derive the index spec for default_sort, e.g. for a DataObject table or for a many_many join table.
+     */
+    public function deriveIndexFromSort(string $tableName, array $fieldNames, string|array $sort, string $indexMode): array
```

Comment:
Logic for this method is just all pulled out of `buildSortDatabaseIndexes()` so we can reuse it in `DataObject` for many_many join tables